### ISSUE TITLE
Transformer decoder block

### ISSUE THREAD

**Issue by s26435** [Science-Club-UG-BioAI/Nightingale#21]
Napisz pojedyńczy blok Transformera do Decodera:

1. Konstruktor:
* Argumenty: `d_model: int`, `nhead: int`, `dim_feedforward: int`, `dropout: float = 0.1`
* Co robi: 
   * tworzy self multihead attention : `nn.MultiheadAttention(d_model, nhead, dropout=dropout, batch_first=True)`
   * tworzy cross multihead attention : `nn.MultiheadAttention(d_model, nhead, dropout=dropout, batch_first=True)`
   *  tworzy sequential: `[nn.Linear(d_model, dim_feedforward), nn.ReLU(), nn.Dropout(dropout), nn.Linear(dim_feedforward, d_model)]`
   * tworzy trzy warstwy `nn.LayerNorm`
   * tworzy dropout
3. Forward:
* argumenty: `x: torch.Tensor`
* Co robi: 
   *  przechodzi przez self att -> dropout + rezyduum -> norm1 -> cross attention -> dropout + rezyduum -> norm2 -> sequential -> dropout + rezyduum -> norm3 -> return
